src/App.jsx

```diff
@@ -14,33 +18,72 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
+  const visibleGoods = [...goodsFromServer].sort((good1, good2) => {
+    switch (sortField) {
+      case ALPHABETICALLY:
+        return good1.toLowerCase().localeCompare(good2.toLowerCase());
+      case LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (reverse) {
+    visibleGoods.reverse();
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortField(ALPHABETICALLY)}
+          type="button"
+          className={`button is-info ${sortField !== ALPHABETICALLY ? 'is-light' : ''}`}
+        >
+          Sort alphabetically
+        </button>
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        <button
+          onClick={() => setSortField(LENGTH)}
+          type="button"
+          className={`button is-success ${sortField !== LENGTH ? 'is-light' : ''}`}
+        >
+          Sort by length
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          onClick={() => setReverse(!reverse)
```

Comment:
There is a missing closing parenthesis for the `setReverse` function call. This could lead to a syntax error. Ensure that the `onClick` handler is properly closed with a parenthesis.